### PR TITLE
[IMP] im_livechat: small name, ui changes

### PR DESCRIPTION
**Pull request by tsm-odoo**
This PR improves several things linked to visitors in the discuss app:
- Change window header to "Visitor #, country"
- Change operator's name to Online Chat name if defined
- Rename "Public user" to "Visitor #", add country instead of website
- Change name of the thread to "Visitor #, country" (left panel + header)
- Remove "Website" from visitor's name (small and big threads)

part of task-3332872

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

File: addons/im_livechat/static/src/core/thread_model_patch.js [hunk 1]
```diff
@@ -30,6 +30,10 @@ patch(Thread.prototype, "im_livechat", {
         return this.type === "livechat" ? false : this._super();
     },
 
+    get correspondents() {
+        return this._super().filter(({ is_bot }) => !is_bot);
+    },
```
[the remark this answers is not in the document]
Comment (tsm-odoo) [reply addons/im_livechat/static/src/core/thread_model_patch.js]: I didn't know that, that's sad! Pretty sure this issue is present elsewhere. Anyway, updated for this PR :+1:

[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/im_livechat/static/src/core/thread_model_patch.js]: Indeed, I recently saw limitations of vuejs `reactive`. OWL's `reactive` is basically the same, therefore we should also be cautious of this limitation!

[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/im_livechat/static/src/core/thread_model_patch.js]: https://vuejs.org/guide/essentials/reactivity-fundamentals.html#limitations-of-reactive
cc @seb-odoo

**Review by alexkuhn — APPROVED** [review]
@robodoo override=ci/security r+
Markup is static html, name of operator is escaped

File: addons/im_livechat/models/chatbot_script_step.py [hunk 2]
```diff
@@ -346,7 +346,8 @@ def _process_step_forward_operator(self, discuss_channel):
             # then post a small custom 'Operator has joined' notification
             discuss_channel._chatbot_post_message(
                 self.chatbot_script_id,
-                Markup('<div class="o_mail_notification">%s</div>') % _('%s has joined', human_operator.partner_id.name))
+                Markup('<div class="o_mail_notification">%s</div>')
+                % _('%s has joined', human_operator.livechat_username or human_operator.partner_id.name))
```
Comment (seb-odoo) [inline addons/im_livechat/models/chatbot_script_step.py:350]: Didn't we have another PR elsewhere that change "has joined" to "joined" and similar sutff?

Comment (tsm-odoo) [reply addons/im_livechat/models/chatbot_script_step.py:350]: It did not involved this message but I guess we could also adapt this one :man_shrugging:

**Pull request closed without merging**